tests/test_current_day_conditions.py

```diff
@@ -18,12 +21,54 @@ def test_current_weather_snowfall(snow_fall: int, rainfall: int, actual: Current
     assert actual.rainfall == rainfall
 
 
-@pytest.mark.parametrize("months, expected", [
-    ([1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12], [9, 10, 11, 13, 14, 15, 15, 15, 13, 12, 10, 9])
-])
-def test_determine_daylength(months: List[int], expected: List[int]):
+@pytest.mark.parametrize("day_number, geographic_latitude, is_winter,"
+                         " polar_location, northern_hemisphere, expected, year", [
+                             (15, 43.073, True, False, True, 9, 2023),
+                             # Madison example, no polar day or night possible
+                             (365, 68, True, True, True, 0, 2023),
+                             # Winter in polar circle (example take from Barrow, Alaska)
+                             (180, 68, False, True, True, 24, 2023),  # Summer in polar circle
+                             (365, -68, True, True, False, 24, 2023),  # Winter in polar circle (south hemisphere)
+                             (180, -68, False, True, False, 0, 2023),  # Sumer in polar circle (south hemisphere)
+                         ])
+def test_determine_daylength(day_number: int, geographic_latitude: float, is_winter: bool, polar_location: bool,
+                             northern_hemisphere: bool, expected: float, year: int) -> None:
     """Tests that correct day length were returned by the corresponding month"""
-    day_length = []
-    for month in months:
-        day_length.append(CurrentDayConditions.determine_daylength(month))
-    assert day_length == expected
+    with patch('RUFAS.current_day_conditions.CurrentDayConditions.calculate_solar_declination_radians',
+               wraps=CurrentDayConditions.calculate_solar_declination_radians) as mocked_radian_calculation:
+        with patch('RUFAS.util.Utility.day_to_month_conversion',
+                   wraps=Utility.day_to_month_conversion) as mocked_month_conversion:
+            if polar_location:
+                if not is_winter:
+                    if northern_hemisphere:
+                        assert CurrentDayConditions.determine_daylength(
+                            day_number, geographic_latitude, year) == expected
+                    else:
+                        assert CurrentDayConditions.determine_daylength(
+                            day_number, geographic_latitude, year) == expected
+                elif is_winter:
+                    if northern_hemisphere:
+                        assert CurrentDayConditions.determine_daylength(
+                            day_number, geographic_latitude, year) == expected
+                    else:
+                        assert CurrentDayConditions.determine_daylength(
+                            day_number, geographic_latitude, year) == expected
+            else:
+                assert expected == pytest.approx(CurrentDayConditions.determine_daylength(
+                    day_number, geographic_latitude, year), 0.1)
```

Comment:
The suggested code should work for all non-polar tests and all summer/winter polar tests, but not polar tests outside of the extremes. Being able to handle any arbitrary, valid test case is a nice to have so it is up to you if you want to make that change.
```suggestion
            observed = CurrentDayConditions.determine_daylength(day_number, geographic_latitude, year)
            tolerance = 0.1 if not polar_location else 0.0
            assert expected == pytest.approx(CurrentDayConditions.determine_daylength(
                    day_number, geographic_latitude, year), tolerance)
```

Reply:
Should it be:
```suggestion
            observed = CurrentDayConditions.determine_daylength(day_number, geographic_latitude, year)
            tolerance = 0.1 if not polar_location else 0.0
            assert expected == pytest.approx(observed, tolerance)
```

Reply:
Thanks, Ed your suggestions are way cleaner. @ew3361zh I think you're right, that's how I'll adapt to this.